### DISCUSSION TITLE
Configurable licence in footer

### DISCUSSION THREAD

**Discussion by LucyGwilliamAdmin** [open-sdg/open-sdg#1538] [Ideas]
UK are customising the site footer in order to be able to show a license (see bottom of any page on [UK platform](https://sdgdata.gov.uk/)).

A lot of websites require this so maybe we should consider adding it in to Open SDG?

Looking around at a few diff sites, where a site does have a license, they always have text but only sometimes have an image, so maybe image shouldn't be compulsory.

Here is the link to the GDS component on which the UK version is based (ignore Crown copyright bit):
https://design-system.service.gov.uk/components/footer/with-meta/index.html

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin While working on this a different approach occurred to me which might be worth considering. Instead of adding configuration settings for this, we could add a blank include file which is intended to be overridden, exactly like we do with the [scripts-custom.html file](https://github.com/open-sdg/open-sdg/blob/1.8.0-dev/_includes/scripts-custom.html). For this purpose, we would call it "footer-custom.html". I started considering this when I realized that this feature would also need a config setting for the alt text of the image, which means that it would need three settings total (image, alt text, text of license).

I admit that the config setting approach has the benefit of requiring less technical expertise (no HTML involved) but I also see some benefits to this blank include file approach:

1. More flexible (eg, can show multiple licenses, multiple images, and other things besides licenses)
2. This approach could be used in many places throughout the platform, so could become a standard way to customize things. Over time this could prevent the site configuration form from growing larger and larger (eg, avoids adding the three new config settings that this ticket would require).
3. Takes advantage of the Jekyll "override" workflow without the downsides of overriding Open SDG stuff

**Reply by LucyGwilliamAdmin** [reply]
@brockfanning I think that would probably be fine. The advantages make sense and it consistent with the approach we used for `scripts` and `head` - will add to talking points for tomorrow's tech call so can check if any objection.

I tested it out and my only comment is that it might be worth adding some space under the custom footer section: http://uk-sdg-feature-branches.s3-website.eu-west-2.amazonaws.com/feature-site-footer-custom-include/

**Reply by brockfanning** [reply]
@LucyGwilliamAdmin One way to get the spacing there would be to make use of Bootstrap's grid system. For example, instead of this content for footer-custom.html:

```
<div id="oglLicense">
  <img alt="Open Government Licence logo" src="{{ site.baseurl }}/assets/img/OGL_logo.png" width="35px">All data is available under the <a href="http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/">Open Government Licence v3.0</a>, except where otherwise stated
</div>
```

You could use:

```
<div class="row">
  <div class="col-xs-12">
    <div id="oglLicense">
      <img alt="Open Government Licence logo" src="{{ site.baseurl }}/assets/img/OGL_logo.png" width="35px">All data is available under the <a href="http://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/">Open Government Licence v3.0</a>, except where otherwise stated
    </div>
  </div>
</div>
```

Of course that means that not only does the user need to understand HTML, but also a bit of Bootstrap's grid system.

We could hardcode the `<div class="row"><div class="col-xs-12">` bit. But there's always a trade-off - by hardcoding that we remove some of the flexibility, in case a user wanted to add something that was not wrapped in a row/col.

Personally I lean towards maximum flexibility (ie, leaving it up to the user to add that extra code), with the thinking that if a user is going to the trouble to add custom HTML then they are probably also prepared to add custom CSS to style whatever they're adding. I also worry that if we hardcode some wrapper code, then in the future we might get a request to remove that hardcoded wrapper, or countries will start overriding the footer.html in order to get rid of the wrapper, which is what we're trying to avoid.

**Comment by brockfanning** [comment]
Also - I put up a pull-request to demonstrate what I'm talking about above: https://github.com/open-sdg/open-sdg/pull/1563
